dadi/frontend/components/FieldString/FieldString.jsx

```diff
@@ -3,23 +3,167 @@
+     */
+    schema: proptypes.object
+  }
+
+  static defaultProps = {
+    error: false,
+    forceValidation: false,
+    value: ''
+  }
+
+  constructor(props) {
+    super(props)
+  }
+
+  componentDidMount() {
+    const {forceValidation, value} = this.props
+
+    if (forceValidation) {
+      this.validate(value)
+    }
+  }
+
+  componentDidUpdate() {
+    const {forceValidation, value} = this.props
+
+    if (forceValidation) {
+      this.validate(value)
+    }
   }
 
   render() {
+    const {error, value, schema} = this.props
+    const publishBlock = schema.publish || {}
+    const type = publishBlock.multiline ? 'multiline' : 'text'
+
     return (
-      <p>String!</p>
+      <div class={styles.field}>
+        <Label
+          error={error}
+          errorMessage={typeof error === 'string' ? error : null}
+          label={schema.label}
+          comment={schema.required ? 'Required' : 'Optional'}
+        >
+          <TextInput
+            onChange={this.handleOnChange.bind(this)}
+            onKeyUp={this.handleOnKeyUp.bind(this)}
+            type={type}
+            value={value}
+          />
+        </Label>
+      </div>
     )
   }
+
+  handleOnChange(event) {
+    const {onChange, schema} = this.props
+
+    if (typeof onChange === 'function') {
+      onChange.call(this, schema._id, event.target.value)
+    }
+  }
+
+  handleOnKeyUp(event) {
+    this.validate(event.target.value)
+  }
+
+  validate(value) {
+    const {error, onError, schema} = this.props
+    const validation = schema && schema.validation
+    const valueLength = typeof value === 'string' ? value.length : 0
+
+    let validationMessage = typeof schema.message === 'string' && schema.message.length ? schema.message : null
+
+    // As per API docs, validation messages are in the format "must be xxx", which
+    // assumes that something (probably the name of the field) will be prepended to
+    // the string to form a final error message. For this reason, we're prepending
+    // the validation message with "This field", but this is something that we can
+    // easily revisit.
+    if (validationMessage) {
+      validationMessage = 'This field ' + validationMessage
+    }
+
+    let hasErrorsAfterValidation = false
+
+    // If the field is required, we add a minLength validation rule,
+    // if there isn't one already.
+    if (schema.required && !validation.minLength) {
+      validation.minLength = 1
+    }
+
+    if (validation) {
+      Object.keys(validation).forEach(validationRule => {
+        switch (validationRule) {
+          case 'minLength':
+            if (valueLength < validation.minLength) {
+              hasErrorsAfterValidation = validationMessage && !schema.required ? validationMessage : true
+            }
+
+            break
+
+          case 'maxLength':
+            if (valueLength > validation.maxLength) {
+              hasErrorsAfterValidation = validationMessage || true
+            }
+
+            break
```

Comment:
I usually indent `break` inside `case`, because technically it's part of the case block (as in, `break` only fires if that particular `case` is the chosen one).